substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -633,4 +640,626 @@ mod e2e {
+		// are submitted.
+		//
+		// The key behavior being tested: missing pages should be treated as empty/default
+		// rather than causing verification failures.
+		ExtBuilder::signed().build_and_execute(|| {
+			roll_to_signed_open();
+			assert_full_snapshot();
+
+			let paged = mine_full_solution().unwrap();
+			let real_score = paged.score;
+			let submitter = 99;
+			let initial_balance = Balances::free_balance(&submitter);
+
+			assert!(
+				paged.solution_pages.len() > 1,
+				"Test requires a multi-page solution, got {} pages",
+				paged.solution_pages.len()
+			);
+
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(submitter), real_score));
+			assert_eq!(
+				balances(submitter),
+				(initial_balance - SignedDepositBase::get(), SignedDepositBase::get())
+			);
+
+			// Submit ONLY page 0 of the solution, deliberately skip pages 1 and 2.
+			let first_page = paged.solution_pages.pagify(Pages::get()).next().unwrap();
+			assert_ok!(SignedPallet::submit_page(
+				RuntimeOrigin::signed(submitter),
+				first_page.0,
+				Some(Box::new(first_page.1.clone()))
+			));
+
+			// Verify the metadata shows correct page submission status
+			let metadata = Submissions::<Runtime>::metadata_of(0, submitter).unwrap();
+			let pages_status = metadata.pages.into_inner();
+			assert_eq!(pages_status[0], true, "Page 0 should be submitted");
+			assert_eq!(pages_status[1], false, "Page 1 should not be submitted");
+			assert_eq!(pages_status[2], false, "Page 2 should not be submitted");
+
+			// Verify that the solution data provider can access all pages without errors
+
+			let page_0 = SignedPallet::get_page(0);
+			let page_1 = SignedPallet::get_page(1);
+			let page_2 = SignedPallet::get_page(2);
+
+			// Page 0 should have actual data, pages 1 and 2 should be empty (default)
+			assert_ne!(page_0, Default::default(), "Submitted page 0 should have data");
+			assert_eq!(page_1, Default::default(), "Missing page 1 should return empty solution");
+			assert_eq!(page_2, Default::default(), "Missing page 2 should return empty solution");
+
+			// Start verification process
+			roll_to_signed_validation_open();
+
+			// The verification should proceed and treat the missing pages as empty
+			roll_next(); // Process page 2 (missing, treated as empty)
+			roll_next(); // Process page 1 (missing, treated as empty)
+			roll_next(); // Process page 0 (submitted with real data)
+
+			// Check that verification handled the missing pages gracefully.
+			// Missing pages should be treated as empty pages without errors.
+			assert_eq!(
+				verifier_events(),
+				vec![
+					crate::verifier::Event::Verified(2, 0), // Page 2 missing, treated as empty
+					crate::verifier::Event::Verified(1, 0), // Page 1 missing, treated as empty
+					crate::verifier::Event::Verified(0, 2), // Page 0 submitted with real data
+					crate::verifier::Event::VerificationFailed(0, FeasibilityError::InvalidScore)
```

Comment:
What is also crucial here is that the second value of the enum gives us a lot of signal --no winners were in page 2 and 1.